lib/streamlit/string_util.py

```diff
@@ -55,3 +59,55 @@ def is_binary_string(inp):
     """Guess if an input bytesarray can be encoded as a string."""
     # From https://stackoverflow.com/a/7392391
     return bool(inp.translate(None, TEXTCHARS))
+
+
+def clean_filename(name: str) -> str:
+    """
+    Taken from https://github.com/django/django/blob/196a99da5d9c4c33a78259a58d38fb114a4d2ee8/django/utils/text.py#L225-L238
+
+    Return the given string converted to a string that can be used for a clean
+    filename. Remove leading and trailing spaces; convert other spaces to
+    underscores; and remove anything that is not an alphanumeric, dash,
+    underscore, or dot.
+    """
+    s = str(name).strip().replace(" ", "_")
```

Comment:
Is there a reason to cast it as a string when type is already a string? I'm not super familiar with python but just wondering.

Reply:
Good question, @willhuang1997 
It is more or less can be avoided here, I think since we use this function only in one place with a string parameter.

But AFAIK specifying type doesn't guarantee you, that code will fail if the wrong type provided, since that type annotation doesn't work in runtime, if let's say user input data can be provided. 

I think it is okay, and a little bit safer to explicitly cast value here.

Reply:
We can probably omit the cast in our codebase as I think mypy gives us the type safety to do so here, but I'd imagine it's written this way simply because that's how it was written in the django codebase this was taken from (see the link in the docstring)